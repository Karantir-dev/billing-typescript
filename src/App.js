import Login from "./componets/Pages/Auntification/Login";
import React from 'react';


function App() {
  return (
    <div className="App">
      <header className="App-header">
       <Login /> 
      </header>
    </div>
  );
}

export default App;
