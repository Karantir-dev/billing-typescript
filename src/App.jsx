import React, { Suspense } from 'react'
import { Routes, Route, BrowserRouter } from 'react-router-dom'
import {
  Loader,
  LoginForm,
  SignupForm,
  PasswordChange,
  PasswordReset,
  PrivateRoute,
  PublicRoute,
  Portal,
} from './Components'

import { Provider } from 'react-redux'
import { PersistGate } from 'redux-persist/integration/react'

import entireStore from './Redux/store'

import {
  AuthPage,
  MainPage,
  AccessLogPage,
  AboutAffiliateProgram,
  SupportPage,
} from './Pages'

import * as route from './routes'

export default function App() {
  return (
    <Provider store={entireStore.store}>
      <PersistGate loading={null} persistor={entireStore.persistor}>
        <BrowserRouter>
          <Suspense fallback={<Loader />}>
            <Routes>
              <Route
                path={route.LOGIN}
                element={
                  <PublicRoute
                    children={<AuthPage children={<LoginForm />} />}
                    restricted
                    redirectTo={route.HOME}
                  />
                }
              />
              <Route
                path={route.REGISTRATION}
                element={
                  <PublicRoute
                    children={<AuthPage children={<SignupForm />} />}
                    restricted
                    redirectTo={route.HOME}
                  />
                }
              />

              <Route
                path={route.RESET_PASSWORD}
                element={
                  <PublicRoute
                    children={<AuthPage children={<PasswordReset />} />}
                    restricted
                    redirectTo={route.HOME}
                  />
                }
              />
              <Route
                path={route.CHANGE_PASSWORD}
                element={
                  <PublicRoute
                    children={<AuthPage children={<PasswordChange />} />}
                    restricted
                    redirectTo={route.HOME}
                  />
                }
              />
              <Route
                path={route.HOME}
                element={
                  <PrivateRoute children={<MainPage />} redirectTo={route.LOGIN} />
                }
              />
              <Route
                path={route.ACCESS_LOG}
                element={<PrivateRoute children={<AccessLogPage />} />}
                redirectTo={route.LOGIN}
              />
              <Route
                path={route.SUPPORT}
                element={<PrivateRoute children={<SupportPage />} />}
                redirectTo={route.LOGIN}
              >
                <Route
                  path={':path'}
                  element={<PrivateRoute children={<SupportPage />} />}
                />
              </Route>
              <Route
                path={route.AFFILIATE_PROGRAM_ABOUT}
                element={
                  <PrivateRoute
                    children={<AboutAffiliateProgram />}
                    redirectTo={route.LOGIN}
                  />
                }
              />
            </Routes>
          </Suspense>

          <Portal>
            <Loader />
          </Portal>
        </BrowserRouter>
      </PersistGate>
    </Provider>
  )
}
