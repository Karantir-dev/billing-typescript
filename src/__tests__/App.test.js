import React from 'react'
import App from '../App'


describe('App', () => {
  it('Render without crashing', () => {
    const component = shallow(<App />)
    // expect(component).toMatchSnapshot();
  })
})

