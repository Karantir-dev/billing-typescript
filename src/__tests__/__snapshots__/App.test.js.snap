// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render without crashing 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <BrowserRouter>
      <Suspense
        fallback={<Loader />}
      >
        <Routes>
          <Route
            element={
              <PublicRoute
                redirectTo="/"
                restricted={true}
              >
                <AuthPage>
                  <LoginForm />
                </AuthPage>
              </PublicRoute>
            }
            path="/login"
          />
          <Route
            element={
              <PublicRoute
                redirectTo="/"
                restricted={true}
              >
                <AuthPage>
                  <SignupForm />
                </AuthPage>
              </PublicRoute>
            }
            path="/signup"
          />
          <Route
            element={
              <PublicRoute
                redirectTo="/"
                restricted={true}
              >
                <AuthPage>
                  <PasswordReset />
                </AuthPage>
              </PublicRoute>
            }
            path="/reset_pass"
          />
          <Route
            element={
              <PublicRoute
                redirectTo="/"
                restricted={true}
              >
                <AuthPage>
                  <PasswordChange />
                </AuthPage>
              </PublicRoute>
            }
            path="/change_pass"
          />
          <Route
            element={
              <PrivateRoute
                redirectTo="/login"
              >
                <MainPage />
              </PrivateRoute>
            }
            path="/"
          />
          <Route
            element={
              <PrivateRoute>
                <MainPage />
              </PrivateRoute>
            }
            path="/access_log"
          />
        </Routes>
      </Suspense>
      <Component>
        <Loader />
      </Component>
    </BrowserRouter>
  </PersistGate>
</Provider>
`;
